data/blog/electron-6-0.md

```diff
@@ -0,0 +1,108 @@
+ * webFrame.executeJavaScriptInIsolatedWorld() [#17312](https://github.com/electron/electron/pull/17312)
+ * webviewTag.executeJavaScript() [#17312](https://github.com/electron/electron/pull/17312)
+
+These functions now have two forms, synchronous and Promise-based asynchronous:
+ * `dialog.showMessageBox()`/`dialog.showMessageBoxSync()` [#17298](https://github.com/electron/electron/pull/17298)
+ * `dialog.showOpenDialog()`/`dialog.showOpenDialogSync()` [#16973](https://github.com/electron/electron/pull/16973)
+ * `dialog.showSaveDialog()`/`dialog.showSaveDialogSync()` [#17054](https://github.com/electron/electron/pull/17054)
+
+These functions now return Promises:
+ * `app.dock.show()` [#16904](https://github.com/electron/electron/pull/16904)
+
+### `Electron Helper (Render).app` and `Electron Helper (Plugin).app`
+
+In order to enable the [hardened runtime](https://developer.apple.com/documentation/security/hardened_runtime_entitlements?language=objc), which restricts things like
+writable-executable memory and loading code signed by a different Team
+ID, special code signing entitlements needed to be granted to the Helper.
+
+To keep these entitlements scoped to the process types that require them, Chromium [added](https://chromium-review.googlesource.com/c/chromium/src/+/1627456) 
+two new variants of the Helper app: one for renderers (`Electron Helper (Render).app`) and one for plugins (`Electron Helper (Plugin).app`).
+
+Folks using `electron-osx-sign` to codesign their Electron app shouldn't have to make any changes to their build logic.
+If you're codesigning your app with custom scripts, you should ensure
+that the two new Helper applications are correctly codesigned.
+
+In order to package your application correctly with these two new helpers you need to be using `electron-packager@14.0.3` or higher.  If you are using `electron-builder` you should follow [this issue](https://github.com/electron-userland/electron-builder/issues/4104) to track support for these new helpers.
+
+## Breaking Changes
+
+ * This release begins laying the groundwork for a future requirement that native Node modules loaded in the renderer process be either [N-API](https://nodejs.org/api/n-api.html) or [Context Aware](https://nodejs.org/api/addons.html#addons_context_aware_addons). The reasons for this change are faster performance, stronger security, and reduced maintenance workload. Read the full details including the proposed timeline in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to be completed in Electron v11.
+
+ * `http.IncomingMessage` headers have [changed slightly](https://github.com/electron/electron/pull/17517#issue-263752903) to more closely match [Node.js behavior](https://nodejs.org/api/http.html#http_message_headers), particularly with the value of `set-cookie` and how duplicate headers are handled. [#17517](https://github.com/electron/electron/pull/17517).
```

Comment:
`net.IncomingMessage` right? we didn't touch the `http` module here

Reply:
good catch. fixed 2765843

￼
￼
￼
￼
￼
 Lock conversation